### PR TITLE
Opgops 1824

### PR DESCRIPTION
**Pull request by brettminnie**
* RDS Automatic backup is now a variable
 * RDS Manual snapshot on delete is now a variable
 * ASG instance count hard set to 0 before we attempt to delete the instance/asg so we don't have race conditions

### CODE REVIEW

File: roles/ec2-app/tasks/destroy_rds_instance.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+
+- name: Destroy RDS instances for stack
+  rds:
+    command: delete
+    instance_name: "{{ rds_instance.db_name + '-' + opg_data.stack  }}"
+    snapshot: "{{ rds_instance.snapshot_name }}"
```
Comment (carinadigital) [inline roles/ec2-app/tasks/destroy_rds_instance.yml:7]: Are we worried about this snapshot overwriting something we want?

File: roles/ec2-app/tasks/destroy_rds_instance.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+- block:
+  - name: Wait until RDS instance is destroyed
+    rds:
+      command: facts
+      instance_name: "{{ rds_instance.db_name + '-' + opg_data.stack }}"
+    register: rds_instance_facts
+    until: rds_instance_facts.instance.status != "deleting"
+    retries: 30
+    delay: 60
+    when: rds_destroy|changed
```
Comment (carinadigital) [inline roles/ec2-app/tasks/destroy_rds_instance.yml:19]: does where does `rds_destroy` variable get updated from?

File: roles/ec2-app/tasks/destroy_rds_instance.yml (hunk 1 above)
Comment (brettminnie) [reply roles/ec2-app/tasks/destroy_rds_instance.yml:7]: No, on dev we don't want it to happen, or maybe we do, so if the name is not set, ie ansible `omit` it doesn't create the snapshot

File: roles/ec2-app/tasks/destroy_rds_instance.yml (hunk 2 above)
Comment (brettminnie) [reply roles/ec2-app/tasks/destroy_rds_instance.yml:19]: Scroll up, there is a register in the task before this

File: roles/ec2-app/tasks/destroy_stack.yml [hunk 3]
```diff
@@ -123,6 +102,12 @@
     state: absent
   with_items: "{{ sns_topics|default([]) }}"
 
+- name: Nullificate ASG instances
```
Comment (carinadigital) [inline roles/ec2-app/tasks/destroy_stack.yml:105]: Better wording, maybe? Nullificate > empty

Comment (brettminnie) [reply roles/ec2-app/tasks/destroy_stack.yml:105]: image

File: roles/ec2-app/tasks/destroy_rds_instance.yml (hunk 2 above)
Comment (carinadigital) [reply roles/ec2-app/tasks/destroy_rds_instance.yml:19]: So asynchronously gets updated?

**Comment by carinadigital** [comment]
Any protection for prod stacks?

**Comment by brettminnie** [comment]
It's set in metadata, has been tested insofar as we've made sure the rds
snapshot name is generated and not an omit function reference

**Comment by carinadigital** [comment]
I mean the RDS deletion.

**Comment by brettminnie** [comment]
If we're deleting the stack we're deleting rds

Why is this closed?

**Pull request merged**